### ISSUE TITLE
Improvement of docstring for Collection or fiona.open

### ISSUE THREAD

**Issue by fnorf** [Toblerity/Fiona#257]
Hey, i just spent WAY too long on a trivial problem. First time Fiona/Shapely and actually also first time using Python to do anything a bit more sophisticated than a script.

I did not realise that the fiona.open Collection was "emptying" as I was looping over it. So I used two nested loops and wondered why it just would not work. The outer loop was supposed to look at each record once, the inner loop used the same collection and compared each record with the current record of the outer loop. Naturally, this lead to very confusing behaviour.

I come from Java with JTS and OpenJUMP where a FeatureCollection would have work as I thought this worked.

I am using PyCharm, the docstring for Collection just told me that it is "A file-like interface to features in the form of GeoJSON-like mappings" (as it says on http://toblerity.org/fiona/fiona.html).

If I had known that it was an iterator, I would probably have spotted my mistake earlier. As I said, I am new to Python, so this should be a trivial thing but the rest was easy, this was just unexpected.

Long text, what I mean is: Could you add some line to the Collection (or fiona.open?) saying that it iterates this way? :)

**Comment by geowurster** [comment]
@fnorf The `Collection()` docstring could certainly be improved but I think your confusion may be a by-product of being new to Python.  References to a `file-like object` are fairly common in Python documentation and imply a certain set of features and behaviors, one of which is that the object is an iterator.

If you're not familiar, 

By saying `file-like object`, we're assuming you're already familiar with the object returned by Python's builtin `open()`, which is what `Collection()` is modeled after.

In other words, we want this:

```python
with fiona.open(input_file) as src:
    for feature in src:
        # Do something with a feature
```

to be familiar and behave like this:

```python
with open(input_file) as f:
    for line in f:
        # Do something with a line
```

**Comment by sgillies** [comment]
Thanks for the suggestion, @fnorf. Here's what you'll see in 1.6.3:

```
Help on class Collection in module fiona.collection:

class Collection(__builtin__.object)
 |  A file-like interface to features of a vector dataset
 |
 |  Python text file objects are iterators over lines of a file. Fiona
 |  Collections are similar iterators (not lists!) over features
 |  represented as GeoJSON-like mappings.
 |
 |  Methods defined here:
...
```

**Comment by geowurster** [comment]
@sgillies The Fiona docs on Toblerity.org are out of date as well.  What would it take to do an update?

**Comment by fnorf** [comment]
I am late, but thank you very much!
